jupyterlab_github/__init__.py

```diff
@@ -14,47 +14,77 @@
-    client_secret = Unicode('', config=True,
-        help='The Client secret for the GitHub OAuth app')
+    api_url = Unicode(
+        'https://api.github.com', config=True,
+        help="The url for the GitHub api"
+    )
+    access_token = Unicode(
+        '', config=True,
+        help=(
+            "A personal access token for GitHub. If specified it takes "
+            "precedence over the `client_id` and `client_secret`"
+        )
+    )
+    client_id = Unicode(
+        '', config=True,
+        help="The Client ID for the GitHub OAuth app"
+    )
+    client_secret = Unicode(
+        '', config=True,
+        help="The Client secret for the GitHub OAuth app"
+    )
+    validate_cert = Bool(
+        True, config=True,
+        help=(
+            "Whether to validate the servers' SSL certificate on requests "
+            "made to the GitHub api. In general this is a bad idea so only "
+            "disable SSL validation if you know what you are doing!"
+        )
+    )
+
 
 class GitHubHandler(APIHandler):
     """
     A proxy for the GitHub API v3.
 
-    The purpose of this proxy is to add the 'client_id' and 'client_secret'
-    tokens to the API request, which allows for a higher rate limit.
-    Without this, the rate limit on unauthenticated calls is so limited as
-    to be practically useless.
+    The purpose of this proxy is to provide authentication to the API requests
+    which allows for a higher rate limit. Without this, the rate limit on
+    unauthenticated calls is so limited as to be practically useless.
     """
     @gen.coroutine
-    def get(self, path = ''):
+    def get(self, path=''):
         """
-        Proxy API requests to GitHub, adding 'client_id' and 'client_secret'
-        if they have been set.
+        Proxy API requests to GitHub, adding authentication parameter(s) if
+        they have been set.
         """
 
         # Get access to the notebook config object
         c = GitHubConfig(config=self.config)
         try:
-            api_path = url_path_join(GITHUB_API, url_escape(path))
-            # If the config has client_id and client_secret set,
-            # apply them to the request.
-            if c.client_id != '' and c.client_secret != '':
-                api_path = url_concat(api_path,
-                    {'client_id': c.client_id,\
-                     'client_secret': c.client_secret,\
-                     'per_page': 100})
+            api_path = url_path_join(c.api_url, url_escape(path))
+            params = {}
+            if c.access_token != '':
+                # Preferentially use the access_token if set
+                params['access_token'] = c.access_token
+            elif c.client_id != '' and c.client_secret != '':
+                # Otherwise use client_id and client_secret if set
+                params['client_id'] = c.client_id
+                params['client_secret'] = c.client_secret
+            if params:
```

Comment:
Is this `if` statement necessary? Can we initialize params with `per_page` on line 72, and then unconditionally `url_concat` here?